### PR TITLE
Add a `print-schema-and-capabilities` command

### PR DESCRIPTION
**Pull request by daniel-chambers**
This adds a `print-schema-and-capabilities` command that will print a JSON object to stdout that contains the NDC schema and capabilities. This allows easier access to the connector schema and capabilities without needing to boot up the webserver and query the `/schema` and `/capabilities` endpoints.

Example:

```
> cargo run -- print-schema-and-capabilities --configuration .
{"schema":{"scalar_types":{},"object_types":{},"collections":[],"functions":[],"procedures":[]},"capabilities":{"version":"0.1.6","capabilities":{"query":{"nested_fields":{},"exists":{}},"mutation":{}}}}
```

The `print_schema_and_capabilities` is also made public from the crate so that third parties can use it separately from the default_main webserver if they want.

### CODE REVIEW

File: crates/sdk/src/default_main.rs [hunk 1]
```diff
@@ -404,6 +415,53 @@ async fn post_query<C: Connector>(
...
+
+    print_json_schema_and_capabilities(schema, capabilities)?;
+
+    Ok(())
+}
+
+/// This foulness manually writes out a JSON object with schema and capabilities properties.
+/// We do it like this to avoid having to deserialize and reserialize any
+/// JsonResponse::Serialized values.
+fn print_json_schema_and_capabilities(
```
Comment (danieljharvey) [inline crates/sdk/src/default_main.rs]: Is it worth a test that runs this on on the example connector and makes sure that what is generated it at least parseable JSON?

**Review by danieljharvey — APPROVED** [review]

File: crates/sdk/src/default_main.rs (hunk 1 above)
Comment (daniel-chambers) [reply crates/sdk/src/default_main.rs]: Good idea.

Comment (daniel-chambers) [reply crates/sdk/src/default_main.rs]: @danieljharvey Done.

**Comment by daniel-chambers** [comment]
I'm probably gonna rework this and split the sdk crate so that connector CLI plugins can reference the new function without having to pull in the rest of the code, which tends to come with OpenSSL which we don't want to link to in CLI plugins.

**Comment by daniel-chambers** [comment]
Okay, so now if you only reference the ndc-sdk-core crate and disable default features, you won't get any axum and openssl in your deps. Unfortunately a feature flag was still required because our types implement traits from Axum and the no orphan instances rule prevents me from defining those in the ndc-sdk crate.

**Review by danieljharvey — APPROVED** [review]
Splitting the SDK "concepts" from the `default_main` implementation feels quite natural, I'm into this. Since the feature flag adds/removes `axum` completely it feels reasonably solid in my mind? If we added something in future that might bring in `openssl` and friends, like `reqwest`, it'd naturally go in the SDK rather than the core.

**Comment by daniel-chambers** [comment]
We're shelving this for now while the CLI team attempt to improve the existing Docker-based schema/capabilities support rather than switch to this.

**Pull request merged**